niap-cc/Permissions/Tester/app/src/androidTest/java/com/android/certifications/niap/permissions/ReadMediaVisualUserSelectedTest.java

```diff
@@ -0,0 +1,216 @@
+        mUiAutomation = InstrumentationRegistry.getInstrumentation().getUiAutomation();
+        mContext = InstrumentationRegistry.getInstrumentation().getTargetContext();
+        mDevice = UiDevice.getInstance(mInstrumentation);
+    }
+
+    @After
+    public void tearDown() {
+        mUiAutomation.dropShellPermissionIdentity();
+    }
+
+
+    private Activity getActivity() {
+        Activity activity = null;
+
+        Context context = mInstrumentation.getContext();
+        Instrumentation.ActivityMonitor monitor =
+                mInstrumentation.addMonitor(
+                        "com.android.certifications.niap.permissions.activities.MainActivity",
+                        null, false);
+
+        Intent intent = context.getPackageManager()
+                .getLaunchIntentForPackage("com.android.certifications.niap.permissions");
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
+        context.startActivity(intent);
+        try {
+            Thread.sleep(2000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        activity = monitor.waitForActivityWithTimeout(2000);
+
+        return activity;
+    }
+    @Test
+    public void runPermissionTests(){
+
+        boolean hasPermissionInManifest = PermissionUtils.ensureRequiredPermissions(
+                new String[]{android.Manifest.permission.READ_MEDIA_VISUAL_USER_SELECTED},mInstrumentation.getTargetContext()
+        );
+        boolean ok = true;
+        try {
+            MainActivity activity = (MainActivity) getActivity();
+
+            activity.addLogLine(
+                    "[READ_MEDIA_VISUAL_USER_SELECTED Test case]. Please run this test case with no-perm/normal apk variants.");
+
+            //Eval variant and determine expected result//
+            List<String> not_granted = new ArrayList<String>();
+            if (checkSelfPermission(mContext,android.Manifest.permission.READ_MEDIA_VISUAL_USER_SELECTED)
+                    != PackageManager.PERMISSION_GRANTED) {
+                not_granted.add(android.Manifest.permission.READ_MEDIA_VISUAL_USER_SELECTED);
+                not_granted.add(Manifest.permission.READ_MEDIA_IMAGES);
+            }
+
+            //Wait dialogue shown and click "Select phtos and videos"
+            //1-1.Wait chooser dialogue and then choose first one
+            //1-2.Wait chooser dialgoue and then choose nothing
+            //2. close chooser/
+            String[] perms = not_granted.toArray(new String[0]);
+            if(perms.length>0)
+                ActivityCompat.requestPermissions(activity, perms,
+                        Constants.PERMISSION_CODE_RUNTIME_DEPENDENT_PERMISSIONS);
+            else
+                Log.d("tag","the permission is already granted.");
+            mDevice.waitForIdle();
+            UiObject2 btnUserSelected = mDevice.findObject(By.text("Select photos and videos"));
```

Comment:
Add a comment about the locale settings in source code.

I think it's difficult to refer the translated text from resource.
Because the value is included in MediaProvider package (packages/providers/MediaProvider/res/). 
These values are not exposed.